frontend/public/components/factory/list.jsx

```diff
@@ -209,9 +210,61 @@ export const WorkloadListHeader = props => <ListHeader>
   <ColHead {...props} className="col-lg-3 hidden-md hidden-sm hidden-xs" sortField="spec.selector">Pod Selector</ColHead>
 </ListHeader>;
 
-const Rows = ({data, expand, Row, kindObj}) => <div className="co-m-table-grid__body">
-  {data.map(obj => <Row key={obj.rowKey || obj.metadata.uid} obj={obj} expand={expand} kindObj={kindObj} />)}
-</div>;
+export class Rows extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.measurementCache = new CellMeasurerCache({
+      fixedWidth: true,
+      minHeight: 50,
+      keyMapper: rowIndex => this.props.data[rowIndex].metadata.uid,
+    });
+
+    this.rowRenderer = this._rowRenderer.bind(this);
+  }
+
+  _rowRenderer({index, style, key, parent}) {
+    const {data, expand, Row, kindObj} = this.props;
+    const obj = data[index];
+
+    return <CellMeasurer
+      cache={this.measurementCache}
+      columnIndex={0}
+      key={key}
+      rowIndex={index}
+      parent={parent}>
+      <div style={style}>
+        <Row key={obj.metadata.uid} obj={obj} expand={expand} kindObj={kindObj} index={index} />
+      </div>
+    </CellMeasurer>;
+  }
+
+  render() {
+    return <div className="co-m-table-grid__body">
+      <WindowScroller>
+        {({height, isScrolling, registerChild, onChildScroll, scrollTop}) =>
```

Comment:
Slightly confused about `<WindowScroller>` child being an arrow function.  

Fishing for the WidowScroller docs on that.

Reply:
This is the [Function-as-Child pattern](https://medium.com/merrickchristensen/function-as-child-components-5f3920a9ace9), which is superior to traditional `props.children` because it is more extensible and does not "magically" inject `props` (similar problem to "transclusion" in AngularJS). `WindowScroller` and friends are HOCs, but this is how you use them declaratively in JSX/TSX.

Reply:
Huh, that's a little mind-blowing.  Noodling on it a bit.